packages/slider/index.tsx

```diff
@@ -0,0 +1,455 @@
+}
+
+interface SliderState {
+  classList: Set<string>;
+  markerValue?: number;
+  markerStyles: React.CSSProperties[];
+}
+
+class Slider extends React.Component<SliderProps, SliderState> {
+  get classes() {
+    const {classList} = this.state;
+    const {className, discrete, displayMarkers} = this.props;
+    return classnames('mdc-slider', Array.from(classList), className, {
+      'mdc-slider--discrete': discrete,
+      'mdc-slider--display-markers': discrete && displayMarkers,
+    });
+  }
+
+  get adapter(): MDCSliderAdapter {
+    return {
+      /**
+       * Returns true if className exists for the slider Element
+       */
+      hasClass: (className: string): boolean =>
+        this.classes.split(' ').includes(className),
+      /**
+       * Adds a class to the slider Element
+       */
+      addClass: (className: string): void => {
+        const {classList} = this.state;
+        classList.add(className);
+        this.setState({classList});
+      },
+      /**
+       * Removes a class from the slider Element
+       */
+      removeClass: (className: string): void => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      /**
+       * Returns a string if attribute name exists on the slider Element, otherwise returns null
+       */
+      getAttribute: (name: string): string | null => {
+        if (!this.sliderElement.current) {
+          return null;
+        }
+        return this.sliderElement.current.getAttribute(name);
+      },
+      /**
+       * Sets attribute name on slider Element to value
+       */
+      setAttribute: (name: string, value: string): void => {
+        if (!this.sliderElement.current) {
+          return;
+        }
+        this.sliderElement.current.setAttribute(name, value);
+      },
+      /**
+       * Removes attribute name from slider Element
+       */
+      removeAttribute: (name: string): void => {
+        if (!this.sliderElement.current) {
+          return;
+        }
+        this.sliderElement.current.removeAttribute(name);
+      },
+      /**
+       * Returns the bounding client rect for the slider Element
+       */
+      computeBoundingRect: (): ClientRect => {
+        if (!this.sliderElement.current) {
+          // new DOMRect is not IE11 compatible
+          const defaultDOMRect = {
+            bottom: 0,
+            height: 0,
+            left: 0,
+            right: 0,
+            top: 0,
+            width: 0,
+            x: 0,
+            y: 0,
+          };
+          return defaultDOMRect;
+        }
+        return this.sliderElement.current.getBoundingClientRect();
+      },
+      /**
+       * Returns the tab index of the slider Element
+       */
+      getTabIndex: (): number => this.props.tabIndex || 0,
+      /**
+       * Registers an event handler on the root element for a given event.
+       */
+      registerInteractionHandler: (
```

Comment:
We omit the register/deregister methods in mdc react and just use synthetic event handling. Please see https://github.com/material-components/material-components-web-react/blob/master/packages/floating-label/index.tsx#L94 as an example

Reply:
Do I need to re-implement [these handlers](https://github.com/material-components/material-components-web/blob/master/packages/mdc-slider/foundation.ts#L147-L153) in React?